TextFormat.md

```diff
@@ -43,43 +38,379 @@ support more human-readable representations, but never at the cost of accurate r
+     var $1:f32;
+     $1 = f32.reinterpret/i32(1597463007 - ((i32.reinterpret/f32($0)) >> 1));
+     $1 = $1 * (0x1.8p0 - $1 * ($0 = $0 * 0x1p-1) * $1);
+     $1 * (0x1.8p0 - $1 * $0 * $1)
+   }
+```
+
+This shows off the compactness of infix operators with overloading. In the
+s-expression syntax, these expressions are quite awkward to read, and this
+isn't even a very big example. But the text syntax here is very short.
+
+### Labels
+
+Excerpt from labels.wast:
+
+```
+  (func $loop3 (result i32)
+    (local $i i32)
+    (set_local $i (i32.const 0))
+    (loop $exit $cont
+      (set_local $i (i32.add (get_local $i) (i32.const 1)))
+      (if (i32.eq (get_local $i) (i32.const 5))
+        (br $exit (get_local $i))
+      )
+      (get_local $i)
+    )
+  )
+```
+
+Corresponding proposed text syntax:
+
+```
+  function $loop3 () : (i32) {
+    var $i:i32;
+    $i = 0;
+    loop $cont {
+      $i = $i + 1;
+      if ($i == 5) {
+        br ($i) $exit;
+      }
+    $exit:
+    }
+  }
+```
+
+Note that the curly braces are part of the `if`, rather than introducing a
+block. This reflects how `if` essentially provides `block`-like capabilities
+in the wasm binary format.
+
+### Nested blocks
+
+Label definitions, like the `$exit:` above, introduce additional blocks nested
+within the nearest `{`, without requiring their own `{`. This allows the deep
+nesting of `br_table` to be printed in a relatively flat manner:
+
+```
+  {
+    br_table ($index) [$red, $orange, $yellow, $green], $default;
+  $red:
+      // ...
+  $orange:
+      // ...
+  $yellow:
+      // ...
+  $green:
+      // ...
+  $default:
+  }
+```
+
+representing the following in nested form:
+
+```
+  (block $default
+    (block $green
+      (block $yellow
+        (block $orange
+          (block $red
+            (br_table (get_local $index), [$red, $orange, $yellow, $green] $default)
+          )
+          // ...
+        )
+        // ...
+      )
+      // ...
+    )
+    // ...
+  )
+```
+
+`br_table`s can have large numbers of labels, so this feature allows us to
+avoid very deep nesting in many cases.
+
+Note that when a label appears just before the closing `}`, it doesn't introduce
+a new block; it just provides a name for the enclosing block's label.
+
+## Operators with special syntax
+
+The default syntax for operators is their opcode name followed by a
+comma-separated list of operands. However, several operators have
+special syntax.
+
+
+### Control flow operators ([described here](https://github.com/WebAssembly/design/blob/master/AstSemantics.md))
+
+| Name | Syntax | Examples
+| ---- | ---- | ---- |
+| `block` | `{` … *label*: `}` | `{ br $a; a: }`
+| `loop` | `loop` *label* `{` … `}` | `loop $a { br $a; }`
+| `if` | `if` `(` *condition* `)` `{` … `}` | `if (0) { 1 }`
+| `if` | `if` `(` *condition* `) `{` … `} else `{` … `}` | `if (0) { 1 } else { 2 }`
+| `br` | `br` *label* | `br $where`
+| `br` | `br` `(` *expr* `)` *label* | `br ($v) $where`
```

Comment:
I guess you have to remember that `br` doesn't have a condition, while `br_if` does. I grant that this may not be obvious, but `br` having a value operand doesn't have any kind of analog in JS, so there's no obvious precedent to follow here.